clouddriver-kubernetes-v2/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/v2/security/KubernetesV2Credentials.java

```diff
@@ -611,24 +611,22 @@ public void patch(
       String action, List<KubernetesKind> kinds, String namespace, Supplier<T> op) {
     Map<String, String> tags = new HashMap<>();
     tags.put("action", action);
-    if (kinds.size() == 1) {
-      tags.put("kind", kinds.get(0).toString());
-    } else {
-      tags.put(
-          "kinds", kinds.stream().map(KubernetesKind::toString).collect(Collectors.joining(",")));
-    }
+    tags.put(
+        "kinds",
+        kinds.stream().map(KubernetesKind::toString).sorted().collect(Collectors.joining(",")));
     tags.put("account", accountName);
     tags.put("namespace", StringUtils.isEmpty(namespace) ? "none" : namespace);
     tags.put("success", "true");
     long startTime = clock.monotonicTime();
     try {
       return op.get();
     } catch (KubectlException e) {
-      // TODO(ezimanyi): We should be setting success to false here; this does not feel like success
+      tags.put("success", "false");
+      tags.put("reason", e.getClass().getSimpleName());
       throw e;
     } catch (Exception e) {
       tags.put("success", "false");
-      tags.put("reason", e.getClass().getSimpleName() + ": " + e.getMessage());
+      tags.put("reason", e.getClass().getSimpleName());
       throw new KubectlJobExecutor.KubectlException(
```

Comment:
```suggestion
      throw new KubectlException(
```

...or is this different than the un-prefixed `KubectlException` caught above?

Reply:
Oh good call, given that we've imported it already, there's no need to qualify it here.